src/test/Recordings/ActiveRecordingsTable.test.tsx

```diff
@@ -219,45 +222,55 @@ describe('<ActiveRecordingsTable />', () => {
     expect(tree.toJSON()).toMatchSnapshot();
   });
 
-  it('adds a recording after receiving a notification', () => {
-    renderWithServiceContextAndReduxStoreWithRoute(<ActiveRecordingsTable archiveEnabled={true} />, {
-      preloadState: preloadedState,
-      history: history,
+  it('adds a recording after receiving a notification', async () => {
+    await act(async () => {
+      renderWithServiceContextAndReduxStoreWithRoute(<ActiveRecordingsTable archiveEnabled={true} />, {
+        preloadState: preloadedState,
+        history: history,
+      });
     });
     expect(screen.getByText('someRecording')).toBeInTheDocument();
     expect(screen.getByText('anotherRecording')).toBeInTheDocument();
   });
 
-  it('updates the recording labels after receiving a notification', () => {
-    renderWithServiceContextAndReduxStoreWithRoute(<ActiveRecordingsTable archiveEnabled={true} />, {
-      preloadState: preloadedState,
-      history: history,
+  it('updates the recording labels after receiving a notification', async () => {
+    await act(async () => {
+      renderWithServiceContextAndReduxStoreWithRoute(<ActiveRecordingsTable archiveEnabled={true} />, {
+        preloadState: preloadedState,
+        history: history,
+      });
     });
     expect(screen.getByText('someLabel: someUpdatedValue')).toBeInTheDocument();
     expect(screen.queryByText('someLabel: someValue')).not.toBeInTheDocument();
   });
 
-  it('stops a recording after receiving a notification', () => {
-    renderWithServiceContextAndReduxStoreWithRoute(<ActiveRecordingsTable archiveEnabled={true} />, {
-      preloadState: preloadedState,
-      history: history,
+  it('stops a recording after receiving a notification', async () => {
+    await act(async () => {
+      renderWithServiceContextAndReduxStoreWithRoute(<ActiveRecordingsTable archiveEnabled={true} />, {
+        preloadState: preloadedState,
+        history: history,
+      });
     });
     expect(screen.getByText('STOPPED')).toBeInTheDocument();
     expect(screen.queryByText('RUNNING')).not.toBeInTheDocument();
   });
 
-  it('removes a recording after receiving a notification', () => {
-    renderWithServiceContextAndReduxStoreWithRoute(<ActiveRecordingsTable archiveEnabled={true} />, {
-      preloadState: preloadedState,
-      history: history,
+  it('removes a recording after receiving a notification', async () => {
+    await act(async () => {
```

Comment:
Some reasons behind wrapping these calls in `act`?

Reply:
There was a lot of console warnings with the tests before adding that, but other than that not actually sure of the underlying reason why that fixes it.

Reply:
Pretty odd! This only happens with `*RecordingsTable` components :D

Reply:
Seems like using `iframe` for ReportView causes this issue. It uses `onLoad` callback that updates React states outside React callstack, which throws a warning in tests. Surprisingly, this didn't get caught in old versions of React testing.

https://github.com/cryostatio/cryostat-web/blob/19f353f30d8783dae8e7501f75c1a4fbe1cf3311/src/app/Recordings/ReportFrame.tsx#L83